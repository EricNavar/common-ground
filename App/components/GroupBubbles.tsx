import React from "react";
import { TouchableOpacity, StyleSheet, Text, Image, View } from "react-native";
import { Contact } from "../commonTypes";

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
  },
  text: {
    marginRight: "auto",
    marginLeft: "auto",
    width: 100,
  },
  touchable: {
    width: 100,
    height: 100,
    borderRadius: 50,
    backgroundColor: "#ddd",
    position: "relative",
    padding: 20,
    marginBottom: 10,
  },
  image: {
    position: "absolute",
    borderStyle: "solid",
    borderWidth: 3,
    borderColor: "#ddd",
  },
  image0: {
    width: 65,
    height: 65,
    borderRadius: 30,
    top: 10,
  },
  image1: {
    width: 55,
    height: 55,
    borderRadius: 25,
    bottom: 0,
    right: 0,
  },
  image2: {
    width: 45,
    height: 45,
    borderRadius: 20,
    top: 0,
    right: 0,
  },
});

type GroupBubblesProps = {
  members: Contact[];
  name: string;
  onClick: () => void;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  style?: any;
};

function GroupBubbles(props: GroupBubblesProps) {
  return (
    <View style={[styles.container, props.style]}>
      <TouchableOpacity onPress={props.onClick} style={styles.touchable}>
        <Image
          source={{ uri: props.members[0].profilePic }}
          style={[styles.image0, styles.image]}
        />
        {props.members[1] && (
          <Image
            source={{ uri: props.members[1].profilePic }}
            style={[styles.image1, styles.image]}
          />
        )}
        {props.members[2] && (
          <Image
            source={{ uri: props.members[2].profilePic }}
            style={[styles.image2, styles.image]}
          />
        )}
      </TouchableOpacity>
      <Text style={styles.text}>{props.name}</Text>
    </View>
  );
}

export { GroupBubbles };
