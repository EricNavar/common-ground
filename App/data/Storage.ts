import AsyncStorage from '@react-native-async-storage/async-storage';

export const Storage = {
    get: async (key: string) : Promise<string | null> => {
        return await AsyncStorage.getItem(key);
    },
    set: async (key: string, value: string) => {
        await AsyncStorage.setItem(key, value).then(() => {
            console.log('Stored ', value, ' at ', key);
        }).catch((error: any) => {
            console.log('Error storing data at ', key, error);
        });
    },
};