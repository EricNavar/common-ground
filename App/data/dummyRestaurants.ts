export const losPollosHermanos = {
  name: 'Los Pollos Hermanos',
  thumbnail:
    'https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200',
  address: {
    line1: '1515 SW 13th Street',
    line2: 'Gainesville, FL 32601',
  },
  priceRating: 1,
  starRating: 4,
  description:
    'Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.',
  reviews: [
    {
      starRating: 1,
      bodyText: 'It was ok.',
    },
    {
      starRating: 5,
      bodyText:
        'The quesalupa is good. I ordered 5 of them and they were yummy and I ate all of them.',
    },
  ],
};

export const tacoSal = {
  name: 'Taco Sal',
  thumbnail:
    'https://static.wikia.nocookie.net/breakingbad/images/c/c0/TacoSal.png/revision/latest?cb=20191120130106',
  address: {
    line1: '1515 SW 13th Street',
    line2: 'Gainesville, FL 32601',
  },
  priceRating: 2,
  starRating: 3,
  description:
    'Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.',
  reviews: [
    {
      starRating: 1,
      bodyText: 'It was ok.',
    },
    {
      starRating: 5,
      bodyText:
        'The quesalupa is good. I ordered 5 of them and they were yummy and I ate all of them.',
    },
  ],
};

export const loyolasFamilyRestaurant = {
  name: 'Loyola\'s Family Restaurant',
  thumbnail:
    'https://www.breakingbad-locations.com/wp-content/uploads/2014/09/E06-Cornered.mkv_000921613.jpg',
  address: {
    line1: '1515 SW 13th Street',
    line2: 'Gainesville, FL 32601',
  },
  priceRating: 3,
  starRating: 4,
  description:
    'Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.',
  reviews: [
    {
      starRating: 1,
      bodyText: 'It was ok.',
    },
    {
      starRating: 5,
      bodyText:
        'The quesalupa is good. I ordered 5 of them and they were yummy and I ate all of them.',
    },
  ],
};

export const elCamino = {
  name: 'El Camino',
  thumbnail:
    'https://static.wikia.nocookie.net/breakingbad/images/c/cc/ElCamino.JPG/revision/latest?cb=20220503235617',
  address: {
    line1: '1515 SW 13th Street',
    line2: 'Gainesville, FL 32601',
  },
  priceRating: 5,
  starRating: 5,
  description:
    'Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.',
  reviews: [
    {
      starRating: 1,
      bodyText: 'It was ok.',
    },
    {
      starRating: 5,
      bodyText:
        'The quesalupa is good. I ordered 5 of them and they were yummy and I ate all of them.',
    },
  ],
};

export const tacoBell = {
  name: 'Taco Bell',
  thumbnail:
    'https://www.tacobell.com/medias/blog-2019-new-menu-header.jpg?context=bWFzdGVyfGltYWdlc3wxNjYyNzd8aW1hZ2UvanBlZ3xpbWFnZXMvaDFhL2g5OC84OTI4NTEyMTE0NzE4LmpwZ3xjMjA5YzZjNTQyMmM0MWM1ZWE5N2Q1OGQ0NDJjYWVkMzdiNTE0MjMyMjRhMzhmMTZjNGEwYTc4YmMwOGYwOWYx',
  address: {
    line1: '1515 SW 13th Street',
    line2: 'Gainesville, FL 32601',
  },
  priceRating: 4,
  starRating: 1,
  description:
    'Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.',
  reviews: [
    {
      starRating: 1,
      bodyText: 'It was ok.',
    },
    {
      starRating: 5,
      bodyText:
        'The quesalupa is good. I ordered 5 of them and they were yummy and I ate all of them.',
    },
  ],
};

export const allRestaurants = [
  losPollosHermanos,
  tacoSal,
  loyolasFamilyRestaurant,
  elCamino,
  tacoBell,
];

const section1 = {
  sectionTitle: 'Chicken restaurants (not meth)',
  contentData: [
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
  ],
};
const section2 = {
  sectionTitle: 'Local favorites',
  contentData: [
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
  ],
};
const section3 = {
  sectionTitle: 'Late night bites',
  contentData: [
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
  ],
};
const section4 = {
  sectionTitle: 'American',
  contentData: [
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
  ],
};
const section5 = {
  sectionTitle: 'Vegetarian (kind of)',
  contentData: [
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
    losPollosHermanos,
  ],
};
export const exploreSections = [
  section1,
  section2,
  section3,
  section4,
  section5,
];
