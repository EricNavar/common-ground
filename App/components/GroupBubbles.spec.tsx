/* eslint-disable @typescript-eslint/no-empty-function */
import { GroupBubbles } from './GroupBubbles';
import { render } from '@testing-library/react-native';
import React from 'react';
import { allUsers } from '../data/dummyUsers';

const noop = () => { };

describe('GroupBubbles', () => {
    it('Returns nothing for a group with 0 people', () => {
        const result = render(
            <GroupBubbles name="2 member gang" members={[]} onClick={noop} />
        );

        expect(result.toJSON()).toEqual(null);
    });

    it('Renders a GroupBubble with 1 person', () => {
        const members = allUsers.slice(1);
        const result = render(
            <GroupBubbles name="2 member gang" members={members} onClick={noop} />
        );

        expect(result.toJSON()).toMatchSnapshot();
    });

    it('Renders a GroupBubble with 2 people', () => {
        const members = allUsers.slice(2);
        const result = render(
            <GroupBubbles name="2 member gang" members={members} onClick={noop} />
        );

        expect(result.toJSON()).toMatchSnapshot();
    });

    it('Renders a GroupBubble with 3 people', () => {
        const members = allUsers.slice(3);
        const result = render(
            <GroupBubbles name="2 member gang" members={members} onClick={noop} />
        );

        expect(result.toJSON()).toMatchSnapshot();
    });

    it('Renders a GroupBubble with 4 people', () => {
        const members = allUsers.slice(4);
        const result = render(
            <GroupBubbles name="2 member gang" members={members} onClick={noop} />
        );

        expect(result.toJSON()).toMatchSnapshot();
    });

    it('Renders a GroupBubble with 5 people', () => {
        const members = allUsers.slice(5);
        const result = render(
            <GroupBubbles name="2 member gang" members={members} onClick={noop} />
        );

        expect(result.toJSON()).toMatchSnapshot();
    });
});
