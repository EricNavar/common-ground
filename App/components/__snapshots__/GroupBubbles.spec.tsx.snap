// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupBubbles Renders a GroupBubble with 1 person 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ddd",
        "borderRadius": 50,
        "height": 100,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 20,
        "position": "relative",
        "width": 100,
      }
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 22.5,
            "bottom": 5,
            "height": 45,
            "right": 5,
            "width": 45,
          },
          {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#999",
            "display": "flex",
            "justifyContent": "center",
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    >
      <Text>
        +
        3
      </Text>
    </View>
    <Image
      source={
        {
          "uri": "https://external-preview.redd.it/Af65FrXSGlwOobPUnC3MNOI1D3xCCERFWoVFfhzpgio.png?format=pjpg&auto=webp&s=ed7b1feb0f8356ba51ea70f5b26b81c1868bc005",
        }
      }
      style={
        [
          {
            "borderRadius": 27.5,
            "height": 55,
            "right": 5,
            "top": 5,
            "width": 55,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
    <Image
      source={
        {
          "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQS7_zsiM5wfE2uac1B7Xk-jdVGpCKzrQqEwOaK9ak60O1iI4Don3BcWR_sOsumMQetlSw&usqp=CAU",
        }
      }
      style={
        [
          {
            "borderRadius": 30,
            "height": 65,
            "top": 10,
            "width": 65,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": 100,
      }
    }
  >
    2 member gang
  </Text>
</View>
`;

exports[`GroupBubbles Renders a GroupBubble with 2 people 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ddd",
        "borderRadius": 50,
        "height": 100,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 20,
        "position": "relative",
        "width": 100,
      }
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 22.5,
            "bottom": 5,
            "height": 45,
            "right": 5,
            "width": 45,
          },
          {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#999",
            "display": "flex",
            "justifyContent": "center",
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    >
      <Text>
        +
        2
      </Text>
    </View>
    <Image
      source={
        {
          "uri": "https://compote.slate.com/images/fb69a16d-7f35-4103-98c1-62d466196b9a.jpg?width=1200",
        }
      }
      style={
        [
          {
            "borderRadius": 27.5,
            "height": 55,
            "right": 5,
            "top": 5,
            "width": 55,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
    <Image
      source={
        {
          "uri": "https://external-preview.redd.it/Af65FrXSGlwOobPUnC3MNOI1D3xCCERFWoVFfhzpgio.png?format=pjpg&auto=webp&s=ed7b1feb0f8356ba51ea70f5b26b81c1868bc005",
        }
      }
      style={
        [
          {
            "borderRadius": 30,
            "height": 65,
            "top": 10,
            "width": 65,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": 100,
      }
    }
  >
    2 member gang
  </Text>
</View>
`;

exports[`GroupBubbles Renders a GroupBubble with 3 people 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ddd",
        "borderRadius": 50,
        "height": 100,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 20,
        "position": "relative",
        "width": 100,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://static.wikia.nocookie.net/breakingbad/images/4/40/Season_2_-_Skyler.jpg/revision/latest?cb=20090617160516",
        }
      }
      style={
        [
          {
            "borderRadius": 22.5,
            "bottom": 5,
            "height": 45,
            "right": 5,
            "width": 45,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
    <Image
      source={
        {
          "uri": "https://static.wikia.nocookie.net/villains/images/3/38/740full-stan-edgar.jpg/revision/latest/scale-to-width-down/250?cb=20190925225050",
        }
      }
      style={
        [
          {
            "borderRadius": 27.5,
            "height": 55,
            "right": 5,
            "top": 5,
            "width": 55,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
    <Image
      source={
        {
          "uri": "https://compote.slate.com/images/fb69a16d-7f35-4103-98c1-62d466196b9a.jpg?width=1200",
        }
      }
      style={
        [
          {
            "borderRadius": 30,
            "height": 65,
            "top": 10,
            "width": 65,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": 100,
      }
    }
  >
    2 member gang
  </Text>
</View>
`;

exports[`GroupBubbles Renders a GroupBubble with 4 people 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ddd",
        "borderRadius": 50,
        "height": 100,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 20,
        "position": "relative",
        "width": 100,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://static.wikia.nocookie.net/breakingbad/images/4/40/Season_2_-_Skyler.jpg/revision/latest?cb=20090617160516",
        }
      }
      style={
        [
          {
            "borderRadius": 27.5,
            "height": 55,
            "right": 5,
            "top": 5,
            "width": 55,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
    <Image
      source={
        {
          "uri": "https://static.wikia.nocookie.net/villains/images/3/38/740full-stan-edgar.jpg/revision/latest/scale-to-width-down/250?cb=20190925225050",
        }
      }
      style={
        [
          {
            "borderRadius": 30,
            "height": 65,
            "top": 10,
            "width": 65,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": 100,
      }
    }
  >
    2 member gang
  </Text>
</View>
`;

exports[`GroupBubbles Renders a GroupBubble with 5 people 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ddd",
        "borderRadius": 50,
        "height": 100,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 20,
        "position": "relative",
        "width": 100,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://static.wikia.nocookie.net/breakingbad/images/4/40/Season_2_-_Skyler.jpg/revision/latest?cb=20090617160516",
        }
      }
      style={
        [
          {
            "borderRadius": 30,
            "height": 65,
            "top": 10,
            "width": 65,
          },
          {
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 3,
            "position": "absolute",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": 100,
      }
    }
  >
    2 member gang
  </Text>
</View>
`;
